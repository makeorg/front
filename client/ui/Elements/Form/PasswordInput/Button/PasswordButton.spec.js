import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { PasswordButton } from './index';

jest.mock('Client/ui/Elements/ButtonElements', () => ({
  UnstyledButtonStyle: 'UnstyledButtonStyle',
}));

jest.mock('Client/ui/Elements/Form/Styled/Icons', () => ({
  HidePasswordIconStyle: 'HidePasswordIconStyle',
}));

describe('PasswordButton', () => {
  it('must match the snapshot with defaultProps', () => {
    const component = renderer.create(<PasswordButton />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(<PasswordButton tabIndex="-1" />);
    const PositiveTabIndex = renderer
      .create(<PasswordButton tabIndex="0" />)
      .toJSON();
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when password is displayed vs hidden', () => {
    const ShowPassword = renderer
      .create(<PasswordButton passwordIsDisplayed />)
      .toJSON();
    const HidePassword = renderer
      .create(<PasswordButton passwordIsDisplayed={false} />)
      .toJSON();
    expect(snapshotDiff(ShowPassword, HidePassword)).toMatchSnapshot();
  });
});
