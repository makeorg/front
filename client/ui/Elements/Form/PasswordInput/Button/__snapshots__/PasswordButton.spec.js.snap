// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordButton must match the snapshot with defaultProps 1`] = `
<UnstyledButtonStyle
  aria-hidden={true}
  as="UnstyledButtonStyle"
>
  <svg
    icon="eye"
    lib="@fortawesome/react-fontawesome"
    prefix="fas"
  />
</UnstyledButtonStyle>
`;

exports[`PasswordButton must return the diff between snapshot when password is displayed vs hidden 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`PasswordButton must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  <UnstyledButtonStyle
    aria-hidden={true}
    as="UnstyledButtonStyle"
-   tabIndex="-1"
+   tabIndex="0"
  >
    <svg
      icon="eye"
      lib="@fortawesome/react-fontawesome"
      prefix="fas"
`;
